frontend/components/commons/dataset-train/DatasetTrain.component.vue

```diff
@@ -0,0 +1,205 @@
+              target="_blank"
+              >{{ button.linkText }}</base-button
+            >
+          </div>
+        </div>
+      </transition>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    datasetTask: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      selectedComponent: null,
+      currentTaskLibraries: [],
+    };
+  },
+  async fetch() {
+    const libraries = this.getLibraries();
+    this.getCurrentTaskLibraries(libraries).forEach(async (library) => {
+      const newLib = await libraries(library);
+      this.currentTaskLibraries.push(newLib);
+    });
+  },
+  computed: {
+    sortedSnippetsTabs() {
+      return this.snippetsTabs.sort();
+    },
+    visibleTab() {
+      return this.selectedComponent || this.snippetsTabs[0] || null;
+    },
+    snippetsTabs() {
+      return this.currentTaskLibraries.map((library) => {
+        return this.getLibrary(library?.attributes?.title);
+      });
+    },
+    task() {
+      switch (this.datasetTask) {
+        case "TextClassification":
+          return "text-classification";
+        case "TokenClassification":
+          return "token-classification";
+        case "Text2Text":
+          return "text2text";
+      }
+    },
+    parsedSnippet() {
+      const docElement = new DOMParser().parseFromString(
+        this.snippet.html,
+        "text/html"
+      ).documentElement;
+      const preBlocks = docElement.getElementsByTagName("pre");
+      for (let i = 0; i < preBlocks.length; i++) {
+        const code = preBlocks[i].innerText;
+        preBlocks[i].innerHTML = `<base-code code='${code}'></base-code>`;
+      }
+      const html = docElement.getElementsByTagName("body")[0].innerHTML;
+      const htmlWithVariables = html
+        .replace("<my_dataset_name>", this.datasetName)
+        .replace("<my_workspace_name>", this.workspaceName);
+      return `<div>${htmlWithVariables}</div>`;
+    },
+    snippetAttributes() {
+      return this.snippet?.attributes;
+    },
+    snippet() {
+      return (
+        this.currentTaskLibraries.find(
+          (library) => library.attributes.title === this.visibleTab.name
+        ) || {}
+      );
+    },
+    workspaceName() {
+      return this.$route.params.workspace;
+    },
+    datasetName() {
+      return this.$route.params.dataset;
+    },
+  },
+  methods: {
+    getCurrentTaskLibraries(libraries) {
+      return (
+        libraries?.keys().filter((library) => library.includes(this.task)) || []
+      );
+    },
+    getSelectedLibrary(id) {
+      this.selectedComponent = this.snippetsTabs.find(
+        (snippet) => snippet.id === id
+      );
+    },
+    getLibrary(name) {
+      return {
+        id: name.trim().toLowerCase(),
+        name,
+      };
+    },
+    getLibraries() {
+      let libraries = null;
+      try {
+        libraries = require.context(
+          `../../../../docs/_source/_common/snippets/training`,
```

Comment:
it's better to use absolute and not relative import

Reply:
it's outside of the project so I tried to create an alias with webpack for the path, but didn't work

Reply:
Maybe we can define symlink to link and share the same folder in several places

@davidberenstein1957 @leiyre Is there any requirement to have these code snippets in the docs section? Will be shown there also?

If only code snippets are used in the frontend part, I would like to keep them as part of the UI static resources, 
if not, we can even have a symbolic link inside the frontend folder and work as if they were there.